pilot/pkg/bootstrap/server.go

```diff
@@ -795,14 +795,14 @@ func (s *Server) sseMCPController(args *PilotArgs,
 	clients *[]*sink.Client,
 	configStores *[]model.ConfigStoreCache) {
 	clientNodeID := "SSEMCP"
-	s.discoveryOptions = &coredatamodel.DiscoveryOptions{
+	s.incrementalMcpOptions = &coredatamodel.Options{
 		DomainSuffix: args.Config.ControllerOptions.DomainSuffix,
 	}
-	s.mcpDiscovery = coredatamodel.NewMCPDiscovery(s.discoveryOptions)
-	s.incrementalMcpOptions = &coredatamodel.Options{
+	controller := coredatamodel.NewSyntheticServiceEntryController(s.incrementalMcpOptions)
```

Comment:
how does the re-arranging here address the flakes?

Reply:
The rearrangement is because it changes which components keeps track of the NotReadyEndpoints. The tracking of NotReady endpionts is now defferd to controller since it architectrally makes more sense to reside in there an not the discovery.